packages/builder/src/stores/builder/automations.ts

```diff
@@ -1889,6 +1894,7 @@ const automationActions = (store: AutomationStore) => ({
     } else {
       await API.deleteAutomation(automation._id!, automation._rev!)
     }
+    await deploymentStore.publishApp()
```

Comment:
I'd be somewhat worried making the full `publishApp` call everytime something changes with automations/workspace apps, if you follow this function the call to `completePublish` is quite expensive, it does a full application list everytime (`/api/applications?status=all`) - this is a very expensive call and in the Cloud will mean these changes come with a significant lag.

It might be better to either have a new publish app call which is more lightweight, or to do some of these updates in the background - realistically I don't believe we need to fetch all apps everytime we publish, its just updating app metadata/statuses.